Automattic/wp-calypso
# GF Signup: Add a modal upsell when a user does not chose a custom domain and choses a free plan

Related to Automattic/growth-foundations#139

## Proposed Changes

Upsells a personal plan when the user selects a free plan or skips domain selection. And then tries to opt for a free plan.

<img width="709" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3422709/96249b78-a80b-4ca3-9672-ec2ba9f95d7f">


This upsell is currently behind a feature flag. This needs to be wired up with the existing experiment infrastructure which will be done in a subsequent PR.

## Testing Instructions

1. Go to `/start?flags=onboarding-pm%2Ffree-free-modal` or `/start/onboarding-pm?flags=onboarding-pm%2Ffree-free-modal`
2. Either select the free subdomain, or ` Choose my domain later` 
3. Select the free plan
4. The above popup should be visible
5. Any other combination of flows that use domains+plans should remain unaffected.

### From @isatuncman-auto 

#### We want to implement free/free modal. There are two ways users could hit this banner in signup funnel:
 - User select a free domain (or skip domain selection) and choose a free plan --> **Show the free/free modal**
 - User chooses a paid domain + free plan hit the paid/free modal, downgrade to free domain --> **Do not show the free/free modal**



## Pre-merge Checklist
